k3s-io/k3s
# Fix to allow non-root users access to storage volumes.

#### Proposed Changes ####
Non root users can now access local-storage volumes, while making it difficult to view/access all storage volumes. The parent directory `/var/lib/rancher/k3s/storage` is now 701, allowing non-root users access to sub-directories if they know the exact name of the sub-directory. All created local-storage volumes are reverted back to `777`. 

#### Types of Changes ####
Bugfix

#### Verification ####
- Start k3s server
- Create local storage by following rancher guide https://rancher.com/docs/k3s/latest/en/storage/
- Verify that storage directory at  `/var/lib/rancher/k3s/storage/` has 0701 permissions
- Verify that the create sub-directory for the local volume has 777 permissions.
- As a non-root user, cd into the created subdirectory directly.

#### Linked Issues ####
https://github.com/k3s-io/k3s/issues/3704

#### User-Facing Change ####
```release-note
NONE
```

#### Further Comments ####
**Security thru obscurity:** The solution of having the `/var/lib/rancher/k3s/storage` as 701 will not prevent other users from potentially accessing other storage pods. It just prevents easy access via tools like `ls` and such. If someone knows the name of the subdirectory, or can guess it, as the subdir is 777 there is nothing stopping them from accessing it.
